project1/server/server.cpp

```diff
@@ -14,16 +14,16 @@ namespace server {
         std::shared_ptr<server_tasks::CommandIDs> active_ids;
 
         // pass active command list to nPortTask and tPortTask
-        auto nPortTask = std::make_shared<server_tasks::NPortTask>(active_ids,nPort,
-                                                                   read_manager_,write_manager_);
-        auto tPortTask = std::make_shared<server_tasks::TPortTask>(active_ids,tPort,
-                                                                   read_manager_,write_manager_);
+        auto nPortTask = std::make_shared<server_tasks::NPortTask>(std::move(active_ids),nPort,
+                                                                   std::move(read_manager_),std::move(write_manager_));
```

Comment:
Again, I don't think `std::move`ing shared pointer arguments gets you anything. It might be worth reading about C++ move semantics if you're curious.

Reply:
i definitely was getting `unique_ptr` and `shared_ptr` confused, mainly becuase I'm not exactly sure how their functionality differs.